### PR TITLE
use go.uber.com/atomic in pkg/logs

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Uses go.uber.com/atomic to ensure alignment of atomic access in pkg/logs

### Motivation

Alignment issues with atomic access.  See #11716 

### Describe how to test/QA your changes

Test tailing a file with rotation, on both nix and windows.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by alai97 — APPROVED** [review]
Looks good for developer docs!

File: pkg/logs/config/source.go [hunk 1]
```diff
@@ -60,7 +57,7 @@ func NewLogSource(name string, config *LogsConfig) *LogSource {
 		inputs:           make(map[string]bool),
 		lock:             &sync.Mutex{},
 		Messages:         NewMessages(),
-		BytesRead:        expvar.Int{},
```
Comment (Kaderinho) [inline pkg/logs/config/source.go:63]: Why are we not using `*expvar.Int{}` as shown in `atomics.md` ? Are those 2 things equivalent ?

Comment (djmitche) [reply pkg/logs/config/source.go:63]: expvar.Int{} is appropriate if the value is used as an expvar -- but this one isn't.

Comment (djmitche) [reply pkg/logs/config/source.go:63]: (mentioned in the commit message: `This value was of type `expvar.Int{}` but never used as an expvar.`)

Comment (Kaderinho) [reply pkg/logs/config/source.go:63]: Didn't see the message mb, thanks !

File: pkg/logs/internal/tailers/file/tailer.go [hunk 2]
```diff
@@ -289,32 +289,11 @@ func (t *Tailer) forwardMessages() {
...
-// fileHasRotated causes subsequent calls to hasFileRotated to return true.
-func (t *Tailer) fileHasRotated() {
-	atomic.StoreInt32(&t.didFileRotate, 1)
-}
-
-// hasFileRotated returns true if the file has been rotated, and this tailer replaced
-// with a new tailer for the new file.
-func (t *Tailer) hasFileRotated() bool {
-	return atomic.LoadInt32(&t.didFileRotate) != 0
-}
```
[the remark this answers is not in the document]
Comment (djmitche) [reply pkg/logs/internal/tailers/file/tailer.go:316]: I found "fileHasRotated" and "hasFileRotated" too similar to remember which was which .  I guess clarity is subjective :)

To my eye, this is all one "class", just spread over multiple files to work around Go's file-level tagging limitation, so accessing fields directly makes sense.  And, I think it's important to surface the atomicity of these accesses.  For example: it makes it clear that we are doing
```
x = someAtomic.Load()
// .. do some stuff ..
someAtomic.Store(x)
```
which is not atomic at all (imagine that something else modifies someAtomic while "do some stuff" is executing).

So, hopefully I've convinced you and it's OK to keep this as-is.

[the remark this answers is not in the document]
Comment (Kaderinho) [reply pkg/logs/internal/tailers/file/tailer.go:316]: It makes sense to me ! 👍

**Review by Kaderinho — APPROVED** [review]
LGTM

**Review by mx-psi — APPROVED** [review]

**Pull request merged**